test/suites/device-mqtt/config_test.go

```diff
@@ -3,42 +3,31 @@ package test
 import (
 	"edgex-snap-testing/test/utils"
 	"testing"
-
-	"github.com/stretchr/testify/assert"
 )
 
-func setupSubtestConfiguration(t *testing.T) {
-	t.Log("[SUBTEST SETUP]")
-	utils.Exec(t, "sudo snap start --enable edgex-device-mqtt.device-mqtt")
-}
+// Deprecated
+func TestEnvConfig(t *testing.T) {
 
-func TestConfiguration(t *testing.T) {
-	setupSubtestConfiguration(t)
+	t.Run("change service port", func(t *testing.T) {
+		// start clean
+		utils.Exec(t, "sudo snap stop edgex-device-mqtt.device-mqtt")
 
-	t.Cleanup(func() {
-		t.Log("[SUBTEST CLEANUP]")
-		utils.Exec(t, "sudo snap stop --disable edgex-device-mqtt.device-mqtt")
-	})
+		t.Cleanup(func() {
+			utils.Exec(t, "sudo snap stop edgex-device-mqtt.device-mqtt")
+			utils.Exec(t, "sudo snap unset edgex-device-mqtt env.service.port")
+		})
 
-	t.Run("change-the-maximum-startup-duration", func(t *testing.T) {
-		startupDurationValue := "120"
+		const newPort = "56789"
 
-		utils.Exec(t, "sudo snap set edgex-device-mqtt startup-duration="+startupDurationValue)
-
-		stdout, _ := utils.Exec(t, "sudo snap get edgex-device-mqtt startup-duration")
-		assert.Equal(t, startupDurationValue+"\n", stdout, "maximum startup-duration does not set successfully")
+		// make sure the port is available before using it
+		utils.CheckPortAvailable(t, newPort)
```

Comment:
How do you distinguish if the `newPort` will be used by device-mqtt or other services?
Do we need to first make sure `newPort`  is not been used by any service?

Reply:
I think that's exactly what I'm doing on line 23, i.e. checking if the port is available (free) before trying to let the mqtt service use it. If port is used by another server, line 23 will throw an error and tests would stop.

Reply:
`sudo lsof -nPi :%s | true` does't throw me error neither using an used or unused port. What kind of error you got?

Reply:
Good catch. A pipe is missing. It should be `||` (see https://github.com/canonical/edgex-snap-testing/pull/17) In the function implementation, if stdout is empty, it means the port is available.